service/src/integrationTest/kotlin/fi/espoo/evaka/ChildBuilder.kt

```diff
@@ -0,0 +1,280 @@
+            )
+            return childFixture
+        }
+    }
+
+    class BackupCareBuilder(
+        private val tx: Database.Transaction,
+        private val today: LocalDate,
+        private val childFixture: ChildFixture
+    ) {
+        private var unitId: UUID? = null
+        private var groupId: UUID? = null
+        private var from: Int = 0
+        private var to: Int = 0
+
+        fun toUnit(id: UUID) = this.apply { this.unitId = id }
+        fun toGroup(id: UUID) = this.apply { this.groupId = id }
+        fun fromDay(day: Int) = this.apply { this.from = day }
+        fun toDay(day: Int) = this.apply { this.to = day }
+
+        fun execAndReturn(): ChildFixture {
+            tx.insertTestBackUpCare(
+                childId = childFixture.childId,
+                unitId = unitId ?: throw IllegalStateException("unit not set"),
+                startDate = today.plusDays(from.toLong()),
+                endDate = today.plusDays(to.toLong()),
+                groupId = groupId
+            )
+            return childFixture
+        }
+    }
+
+    class PlacementBuilder(
+        private val tx: Database.Transaction,
+        private val today: LocalDate,
+        private val childFixture: ChildFixture
+    ) {
+        private var unitId: UUID? = null
+        private var type: PlacementType? = null
+        private var from: Int = 0
+        private var to: Int = 0
+
+        fun toUnit(id: UUID) = this.apply { this.unitId = id }
+        fun ofType(type: PlacementType) = this.apply { this.type = type }
+        fun fromDay(day: Int) = this.apply { this.from = day }
+        fun toDay(day: Int) = this.apply { this.to = day }
+
+        fun exec(): PlacementFixture {
+            val placementId = tx.insertTestPlacement(
+                childId = childFixture.childId,
+                unitId = unitId ?: throw IllegalStateException("unit not set"),
+                type = type ?: throw IllegalStateException("type not set"),
+                startDate = today.plusDays(from.toLong()),
+                endDate = today.plusDays(to.toLong())
+            )
+            return PlacementFixture(
+                tx = tx,
+                today = today,
+                childFixture = childFixture,
+                placementId = placementId,
+                placementPeriod = FiniteDateRange(today.plusDays(from.toLong()), today.plusDays(to.toLong()))
+            )
+        }
+
+        fun execAndReturn(): ChildFixture {
+            exec()
+            return childFixture
+        }
+    }
+
+    class PlacementFixture(
+        private val tx: Database.Transaction,
+        private val today: LocalDate,
+        val childFixture: ChildFixture,
+        val placementId: UUID,
+        val placementPeriod: FiniteDateRange
+    ) {
+        fun withGroupPlacement() = GroupPlacementBuilder(tx, today, this)
+
+        fun withServiceNeed() = ServiceNeedBuilder(tx, today, this)
+
+        fun returnFromPlacement() = childFixture
```

Comment:
Seems unnecessarily low level having to manually jump between "fixtures".

Reply:
maybe my other comment related to this also answers this one